### PR TITLE
feat: decrypt wallet

### PR DESCRIPTION
**Pull request by Nebulis**

### CODE REVIEW

File: src/commands/wallet/decrypt.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    description: "Path to the wallet",
+    normalize: true,
+  });
+
+export const handler = async (args: DecryptWalletCommand): Promise<void> => {
+  trace(`Args: ${JSON.stringify(args, null, 2)}`);
+  try {
+    const name = uniqueNamesGenerator({ dictionaries: [adjectives, colors, animals], separator: "-" });
+    signale.warn(
+      `You are about to reveal the private key of your wallet. Please type the following word into the terminal to prove that you understand the risks: ${name}`
```
Comment (yehjxraymond) [inline src/commands/wallet/decrypt.ts:27]: What? why need to go through these trouble, even if we want user confirmation, most will simply ask for a y/n, with option to overwrite with `-y` flag. I think this is overengineering

File: src/implementations/utils/wallet.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const getWallet = async ({
   network,
   encryptedWalletPath,
   progress = defaultProgress("Decrypting Wallet"),
-}: WalletOption & NetworkOption & { progress?: (progress: number) => void }): Promise<Wallet> => {
+}: WalletOption & Partial<NetworkOption> & { progress?: (progress: number) => void }): Promise<Wallet> => {
```
Comment (yehjxraymond) [inline src/implementations/utils/wallet.ts:30]: Why is partial needed? seemed like it's not needed...

**Review by yehjxraymond — COMMENTED** [review]
overkill!

File: src/implementations/utils/wallet.ts (hunk 2 above)
Comment (Nebulis) [reply src/implementations/utils/wallet.ts:30]: that's a type is used in many place where it's needed

File: src/commands/wallet/decrypt.ts (hunk 1 above)
Comment (Nebulis) [reply src/commands/wallet/decrypt.ts]: fixed

**Pull request merged**